dot/core/service.go

```diff
@@ -204,12 +210,17 @@ func NewService(cfg *Config) (*Service, error) {
 			transactionQueue: cfg.TransactionQueue,
 			isBabeAuthority:  false,
 			lock:             chanLock,
-			closed:           false,
 			syncLock:         syncerLock,
 			blockNumOut:      cfg.SyncChan,
 			respOut:          respChan,
 		}
 
+		// thread safe way to change closed status
+		canLock := atomic.CompareAndSwapUint32(&srv.closed, 0, 1)
+		if !canLock {
+			panic("[core] Error when trying to change Service status from stopped to started.")
+		}
```

Comment:
duplicated from above, can be moved to below the `else` statement

Reply:
I updated the panic to err, but moving it to after its not really the same. Since the previous block is not the same as it will be calling `babe.NewSession(bsConfig)`, if we move it to the else and do not execute the action before that, it will not have the right logic.